{
  "schema_version": "1.7.0",
  "id": "USN-7599-2",
  "summary": "python-pip vulnerability",
  "details": "USN-7599-1 fixed vulnerabilities in python-urllib3. This update provides\nthe corresponding update for python-pip for CVE-2025-50181.\n\nOriginal advisory details:\n\nJacob Sandum discovered that urllib3 handled redirects even when they were\nexplicitly disabled while using the PoolManager. An attacker could possibly\nuse this issue to obtain sensitive information. (CVE-2025-50181)\n\nIllia Volochii discovered that urllib3 incorrectly handled retry and\nredirect parameters when using Node.js. An attacker could possibly use this\nissue to obtain sensitive information. This issue only affected Ubuntu\n25.04. (CVE-2025-50182)\n",
  "aliases": [],
  "upstream": [
    "UBUNTU-CVE-2025-50181"
  ],
  "related": [],
  "published": "2025-06-26T14:19:38.625465Z",
  "modified": "2025-09-26T09:21:37Z",
  "affected": [
    {
      "package": {
        "ecosystem": "Ubuntu:22.04:LTS",
        "name": "python-pip",
        "purl": "pkg:deb/ubuntu/python-pip@22.0.2+dfsg-1ubuntu0.6?arch=source&distro=jammy"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "22.0.2+dfsg-1ubuntu0.6"
            }
          ]
        }
      ],
      "versions": [
        "20.3.4-4",
        "21.3.1+dfsg-3",
        "22.0.2+dfsg-1",
        "22.0.2+dfsg-1ubuntu0.1",
        "22.0.2+dfsg-1ubuntu0.2",
        "22.0.2+dfsg-1ubuntu0.3",
        "22.0.2+dfsg-1ubuntu0.4",
        "22.0.2+dfsg-1ubuntu0.5"
      ],
      "ecosystem_specific": {
        "binaries": [
          {
            "binary_name": "python3-pip",
            "binary_version": "22.0.2+dfsg-1ubuntu0.6"
          },
          {
            "binary_name": "python3-pip-whl",
            "binary_version": "22.0.2+dfsg-1ubuntu0.6"
          }
        ],
        "availability": "No subscription required"
      },
      "database_specific": {
        "cves_map": {
          "ecosystem": "Ubuntu:22.04:LTS",
          "cves": [
            {
              "id": "CVE-2025-50181",
              "severity": [
                {
                  "type": "CVSS_V3",
                  "score": "CVSS:3.1/AV:N/AC:H/PR:L/UI:N/S:U/C:H/I:N/A:N"
                },
                {
                  "type": "CVSS_V3",
                  "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N"
                },
                {
                  "type": "Ubuntu",
                  "score": "medium"
                }
              ]
            }
          ]
        }
      }
    },
    {
      "package": {
        "ecosystem": "Ubuntu:24.04:LTS",
        "name": "python-pip",
        "purl": "pkg:deb/ubuntu/python-pip@24.0+dfsg-1ubuntu1.2?arch=source&distro=noble"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "24.0+dfsg-1ubuntu1.2"
            }
          ]
        }
      ],
      "versions": [
        "23.2+dfsg-1",
        "23.3+dfsg-1",
        "24.0+dfsg-1",
        "24.0+dfsg-1ubuntu1",
        "24.0+dfsg-1ubuntu1.1"
      ],
      "ecosystem_specific": {
        "binaries": [
          {
            "binary_name": "python3-pip",
            "binary_version": "24.0+dfsg-1ubuntu1.2"
          },
          {
            "binary_name": "python3-pip-whl",
            "binary_version": "24.0+dfsg-1ubuntu1.2"
          }
        ],
        "availability": "No subscription required"
      },
      "database_specific": {
        "cves_map": {
          "ecosystem": "Ubuntu:24.04:LTS",
          "cves": [
            {
              "id": "CVE-2025-50181",
              "severity": [
                {
                  "type": "CVSS_V3",
                  "score": "CVSS:3.1/AV:N/AC:H/PR:L/UI:N/S:U/C:H/I:N/A:N"
                },
                {
                  "type": "CVSS_V3",
                  "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N"
                },
                {
                  "type": "Ubuntu",
                  "score": "medium"
                }
              ]
            }
          ]
        }
      }
    },
    {
      "package": {
        "ecosystem": "Ubuntu:25.04",
        "name": "python-pip",
        "purl": "pkg:deb/ubuntu/python-pip@25.0+dfsg-1ubuntu0.1?arch=source&distro=plucky"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "25.0+dfsg-1ubuntu0.1"
            }
          ]
        }
      ],
      "versions": [
        "24.2+dfsg-1",
        "24.2+dfsg-1ubuntu0.1",
        "24.3.1+dfsg-1",
        "25.0+dfsg-1"
      ],
      "ecosystem_specific": {
        "binaries": [
          {
            "binary_name": "python3-pip",
            "binary_version": "25.0+dfsg-1ubuntu0.1"
          },
          {
            "binary_name": "python3-pip-whl",
            "binary_version": "25.0+dfsg-1ubuntu0.1"
          }
        ],
        "availability": "No subscription required"
      },
      "database_specific": {
        "cves_map": {
          "ecosystem": "Ubuntu:25.04",
          "cves": [
            {
              "id": "CVE-2025-50181",
              "severity": [
                {
                  "type": "CVSS_V3",
                  "score": "CVSS:3.1/AV:N/AC:H/PR:L/UI:N/S:U/C:H/I:N/A:N"
                },
                {
                  "type": "CVSS_V3",
                  "score": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:C/C:L/I:L/A:N"
                },
                {
                  "type": "Ubuntu",
                  "score": "medium"
                }
              ]
            }
          ]
        }
      }
    }
  ],
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://ubuntu.com/security/notices/USN-7599-2"
    },
    {
      "type": "REPORT",
      "url": "https://ubuntu.com/security/CVE-2025-50181"
    }
  ]
}